python_modules/dagster/dagster/_core/execution/context_creation_job.py

```diff
@@ -139,6 +146,7 @@ def create_execution_data(
         scoped_resources_builder=scoped_resources_builder,
         resolved_run_config=context_creation_data.resolved_run_config,
         job_def=context_creation_data.job_def,
+        repository_def_lambda=lambda: context_creation_data.repository_def,
```

Comment:
A bit confused re: the lambda optimization since I'm sure the garbage collector will keep this reference in the lambda closure around?